### PR TITLE
Fix formatting.

### PR DESCRIPTION
**Pull request by vext01**
Fix formatting and improve surrounding support code.

See commits.

### CODE REVIEW

**Comment by vext01** [comment]
Oh hold on. I see a problem with the last commit.

File: .buildbot.sh [hunk 1]
```diff
@@ -14,18 +14,22 @@ cmake -DCMAKE_INSTALL_PREFIX=${INST_DIR} \
...
 make -j `nproc` install
 
+# clang-format any new files that we've introduced ourselves.
+#
+# This requires llvm release tags to be consistent llvm upstream.
+PATH=${INST_DIR}/bin:${PATH}
+git remote add llvm https://github.com/llvm/llvm-project
+git fetch --tags llvm
+sh yk_format_new_files.sh
+git diff --exit-code
```
Comment (ltratt) [inline .buildbot.sh:24]: Since this file is `set -e`, won't this line cause the whole buildbot execution to fail?

Comment (vext01) [reply .buildbot.sh:25]: That's the plan yes. If `clang-format` changed any files, then the formatting is wrong and we don't want to continue.

**Comment by vext01** [comment]
02d2c23 is untested, but CI will tell us if it's right :)

**Comment by vext01** [comment]
splat.

**Comment by vext01** [comment]
bors try

**Comment by ltratt** [comment]
No measurable change I think?

**Comment by vext01** [comment]
32 mins. I can live with it.

OK to squash?

**Comment by ltratt** [comment]
Please go ahead.

**Comment by vext01** [comment]
splat.

**Pull request merged**